cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/DiskFailureDetector.java

```diff
@@ -66,82 +61,70 @@ public DiskFailureDetector(KafkaCruiseControlConfig config,
   }
 
   /**
-   * Skip bad disk detection if any of the following is true:
+   * Skip disk failure detection if any of the following is true:
    * <ul>
-   * <li>Cluster model generation has not changed since the last bad disk check.</li>
-   * <li>There is dead broker in the cluster, {@link BrokerFailureDetector} should take care of the anomaly.</li>
-   * <li>Load monitor is not ready.</li>
-   * <li>There is an ongoing execution.</li>
+   * <li>Cluster model generation has not changed since the last disk failure check.</li>
+   * <li>There are dead brokers in the cluster, {@link BrokerFailureDetector} should take care of the anomaly.</li>
+   * <li>{@link AnomalyDetectorUtils#shouldSkipAnomalyDetection(LoadMonitor, KafkaCruiseControl)} returns true.
    * </ul>
    *
-   * @return True to skip bad disk detection based on the current state, false otherwise.
+   * @return True to skip disk failure detection based on the current state, false otherwise.
    */
-  private boolean shouldSkipBadDiskDetection() {
-    int currentClusterGeneration =  _loadMonitor.clusterModelGeneration().clusterGeneration();
+  private boolean shouldSkipDiskFailureDetection() {
+    int currentClusterGeneration = _loadMonitor.clusterModelGeneration().clusterGeneration();
     if (currentClusterGeneration == _lastCheckedClusterGeneration) {
       if (LOG.isDebugEnabled()) {
-        LOG.debug("Skipping bad disk detection because the model generation hasn't changed. Current model generation {}",
+        LOG.debug("Skipping disk failure detection because the model generation hasn't changed. Current model generation {}",
                   _loadMonitor.clusterModelGeneration());
       }
       return true;
     }
     _lastCheckedClusterGeneration = currentClusterGeneration;
 
-    Set<Integer> deadBrokers = _loadMonitor.deadBrokers(MAX_METADATA_WAIT_MS);
+    Set<Integer> deadBrokers = _loadMonitor.deadBrokersWithReplicas(MAX_METADATA_WAIT_MS);
     if (!deadBrokers.isEmpty()) {
       if (LOG.isDebugEnabled()) {
-        LOG.debug("Skipping goal violation detection because there are dead broker in the cluster, dead broker: {}",
-                  deadBrokers);
+        LOG.debug("Skipping disk failure detection because there are dead broker in the cluster, dead broker: {}",
```

Comment:
Can we take this debug `LOG` out of `if (LOG.isDebugEnabled())`? -- i.e. this has no performance benefit in this case.